src/clj_puppetdb/http.clj

```diff
@@ -112,26 +153,21 @@
   automatically and added to the path."
   ([client :- Client ^String path params]
     {:pre (map? params)}
-    (let [query (if (empty? params)
-                  path
-                  (str path "?" (map->query params)))]
-      #_(println "GET:" query)                                 ;; uncomment this to watch queries
-      (let [{:keys [host opts]} client
-            vcr-dir (:vcr-dir opts)
-            opts (dissoc opts :vcr-dir)
-            response (-> (str host query)
-                         (http-get (assoc opts :as :stream) vcr-dir)
-                         (catching-exceptions {:kind :puppetdb-connection-error}))]
-        (if-not (= 200 (:status response))
-          (throw (ex-info nil {:kind   :puppetdb-query-error
-                               :url    (str host path)
-                               :msg    (slurp (util/make-response-reader response))
-                               :status (:status response)
-                               :params params})))
-        (let [data (-> response
-                       util/make-response-reader
-                       decode-stream-catching-parse-exceptions)
-              headers (:headers response)]
-          [data headers]))))
+    (let [query-info (-> (client)                           ; get client info
+                         (assoc :endpoint path)
+                         (assoc :params   params))
+          response (-> (client client path params)
```

Comment:
As per above, naming the anonymous function in make-client-common removes the need to pass that first argument.